frontend/src/HOC/fetchOnMount.js

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import hoistNonReactStatics from "hoist-non-react-statics";
+import PropTypes from "prop-types";
+import { getOptionsLoaded } from "@/selectors/staticContentSelectors";
+import {
+  fetchMineDisturbanceOptions,
+  fetchStatusOptions,
+  fetchRegionOptions,
+  fetchMineTenureTypes,
+  fetchMineTailingsRequiredDocuments,
+  fetchExpectedDocumentStatusOptions,
+  fetchPermitStatusOptions,
+  fetchApplicationStatusOptions,
+  fetchMineIncidentFollowActionOptions,
+  fetchMineIncidentDeterminationOptions,
+  setOptionsLoaded,
+  fetchProvinceCodes,
+  fetchMineComplianceCodes,
+  fetchVarianceStatusOptions,
+  fetchPartyRelationshipTypes,
+  fetchMineCommodityOptions,
+} from "@/actionCreators/staticContentActionCreator";
+
+/**
+ * @constant FetchOnMount - Higher Order Component that makes all staticContent network requests when Home.js loads.
+ *
+ */
+
+const propTypes = {
+  fetchMineDisturbanceOptions: PropTypes.func.isRequired,
+  fetchStatusOptions: PropTypes.func.isRequired,
+  fetchRegionOptions: PropTypes.func.isRequired,
+  fetchMineTenureTypes: PropTypes.func.isRequired,
+  fetchMineTailingsRequiredDocuments: PropTypes.func.isRequired,
+  fetchExpectedDocumentStatusOptions: PropTypes.func.isRequired,
+  fetchPermitStatusOptions: PropTypes.func.isRequired,
+  fetchApplicationStatusOptions: PropTypes.func.isRequired,
+  fetchMineIncidentFollowActionOptions: PropTypes.func.isRequired,
+  setOptionsLoaded: PropTypes.func.isRequired,
+  fetchProvinceCodes: PropTypes.func.isRequired,
+  fetchMineComplianceCodes: PropTypes.func.isRequired,
+  fetchVarianceStatusOptions: PropTypes.func.isRequired,
+  fetchMineIncidentDeterminationOptions: PropTypes.func.isRequired,
+  fetchMineCommodityOptions: PropTypes.func.isRequired,
+  optionsLoaded: PropTypes.bool.isRequired,
+};
+
+export const FetchOnMount = (WrappedComponent) => {
+  const fetchOnMount = (props) => {
+    if (!props.optionsLoaded) {
+      props.fetchMineDisturbanceOptions();
+      props.fetchStatusOptions();
+      props.fetchRegionOptions();
+      props.fetchMineTenureTypes();
+      props.fetchMineTailingsRequiredDocuments();
+      props.fetchExpectedDocumentStatusOptions();
+      props.fetchPermitStatusOptions();
+      props.fetchApplicationStatusOptions();
+      props.fetchMineIncidentFollowActionOptions();
+      props.fetchProvinceCodes();
+      props.fetchMineComplianceCodes();
+      props.fetchVarianceStatusOptions();
+      props.fetchMineIncidentDeterminationOptions();
+      props.fetchMineCommodityOptions();
+      props.fetchPartyRelationshipTypes();
+      props.setOptionsLoaded();
```

Comment:
Hypothetical question... do we want these all bundled into one, big requirements request like this? Will this continue to grow?

What happens if one of the endpoints is down? If ProvinceCodes never resolve, will `props.optionsLoaded` always be `false`, causing every request to be made on every mount?

I don't think this is a pattern that was introduced in this PR, so there's no need to change it here, but I'd like to know if you/we have given thought to some of the questions I have above.

Reply:
Yes, this will continue to grow, do you have any suggestions to counter that? 

I also did consider the issue if one fails, and now that I think about it, the `props.optionsLoaded` should always be false when `Home.js` mounts for the first time, as nothing should be in the state at this point, so this condition is no longer makes sense. 

Does that sound right to you?